### PR TITLE
SAK-30167 Webservice to set a User Property

### PR DESCRIPTION
**Pull request by ddelblanco**
Very simple webservice to add a userProperty.

http://localhost:8080/sakai-ws/rest/sakai/setUserProperty?sessionid=c3750c88-71d8-4731-9547-47cbd0d9d7d7&eid=admin&key=Company&value=Apereo

if value is null, then the property is deleted
if property exists, then property is updated with the new value.

### CODE REVIEW

File: webservices/cxf/src/java/org/sakaiproject/webservices/SakaiScript.java [hunk 1]
```diff
@@ -3851,6 +3851,49 @@ public String getUserProperty(
...
+
+        try {
+            String userid = userDirectoryService.getUserByEid(eid).getId();
+            UserEdit user = userDirectoryService.editUser(userid);
+            user.getPropertiesEdit().addProperty(key, value);
+            userDirectoryService.commitEdit(user);
+        }
+        catch (Exception e) {
+            LOG.error("WS setUserProperty(): " + e.getClass().getName() + " : " + e.getMessage(), e);
+            return e.getClass().getName() + " : " + e.getMessage();
```
Comment (ern) [inline webservices/cxf/src/java/org/sakaiproject/webservices/SakaiScript.java]: I think it would be better to do one of the following:
1) just return "failure" and log the error typically as a warning (best to do this when there is no exception, or when an exception is not appropriate)
2) let the checked exception bubble (bad, requires more work as exceptions needs to be declared)
3) wrap the checked exception in a RuntimeException (better)
4) create single service exception that can be used everywhere (better, and isolates exceptions from RuntimeExceptions)

So in Sakai you will find the use of 1 and 3, it probably would be nice to have 4 but we don't have it!

Long story short I would change 3891 to return "failure" and change LOG.error to a LOG.warn!

**Comment by ddelblanco** [comment]
Changed following Earle suggestions.

**Comment by ern** [comment]
Thanks Diego looks good, just need to fix the build error missing PreferencesEdit!

**Comment by ddelblanco** [comment]
Oh... I will take a look. It was working and compiling perfectly in my locale... Maybe something has changed related with PreferencesEdit since I contributed this... I'll "rebase" and take a look...

**Comment by ddelblanco** [comment]
Done, compiled in local and working fine.
